indexing-hadoop/src/main/java/io/druid/indexer/hadoop/DatasourceInputFormat.java

```diff
@@ -89,9 +104,11 @@ public int compare(WindowedDataSegment s1, WindowedDataSegment s2)
     List<WindowedDataSegment> list = new ArrayList<>();
     long size = 0;
 
+    JobConf dummyConf = new JobConf();
```

Comment:
How is this supposed to work? The first line of this method says:

```
Configuration conf = context.getConfiguration();
```

However, the values in this configuration are ignored, and a blank "dummy" config is passed on instead. This causes exceptions like described in https://github.com/druid-io/druid/issues/5135

Reply:
I think, you're right . rewriting it to use `org.apache.hadoop.mapreduce.lib.input.FileInputFormat` and supplied JobContext in getSplits(JobContext) arg would probably solve #5135 . Please feel free to submit a PR with patch that does so after its verification.